### PR TITLE
Run the CI test locally

### PR DESCRIPTION
**Pull request by rthallisey**
Give the developer the ability to run the CI test locally. Run the
test by: ```make ci```. It will provision mediawiki and postgresql
apbs then bind them.

Changes proposed in this pull request
 - Ability to run `make run` to test provision & bind

### CODE REVIEW

**Comment by rthallisey** [comment]
[(21) Begin Implementing a CI system to rebuild rpms and containers.](https://trello.com/c/RxiG3NIv/383-21-begin-implementing-a-ci-system-to-rebuild-rpms-and-containers)

File: .travis.yml [hunk 1]
```diff
@@ -34,11 +34,12 @@ script:
...
   - oc create -f ./scripts/broker-ci/mediawiki123.yaml || export ERROR=true
-  - ./scripts/broker-ci/wait-for-pods.sh &> /tmp/wait-for-pods-log || export ERROR=true
+  - ./scripts/broker-ci/wait-for-resource.sh create pod mediawiki >> /tmp/wait-for-pods-log 2>&1 || export ERROR=true
+  - ./scripts/broker-ci/wait-for-resource.sh create pod postgresql >> /tmp/wait-for-pods-log 2>&1 || export ERROR=true
   - sleep 30
   - oc create -f ./scripts/broker-ci/bind-mediawiki-postgresql.yaml || export ERROR=true
   - sleep 5
-  - oc delete pods $(oc get pods -n default | grep mediawiki | awk $'{ print $1 }') -n default || export ERROR=true
+  - oc delete pods $(oc get pods -o name -l app=mediawiki123 -n default | head -1 | cut -f 2 -d '/') -n default
   - sleep 20
```
Comment (jmrodri) [inline .travis.yml]: Is it possible to move most of this to its own script instead of cluttering .travis.yml?

File: .travis.yml [hunk 2]
```diff
@@ -34,11 +34,12 @@ script:
...
   - oc create -f ./scripts/broker-ci/postgresql.yaml || export ERROR=true
   - oc create -f ./scripts/broker-ci/mediawiki123.yaml || export ERROR=true
-  - ./scripts/broker-ci/wait-for-pods.sh &> /tmp/wait-for-pods-log || export ERROR=true
+  - ./scripts/broker-ci/wait-for-resource.sh create pod mediawiki >> /tmp/wait-for-pods-log 2>&1 || export ERROR=true
+  - ./scripts/broker-ci/wait-for-resource.sh create pod postgresql >> /tmp/wait-for-pods-log 2>&1 || export ERROR=true
   - sleep 30
   - oc create -f ./scripts/broker-ci/bind-mediawiki-postgresql.yaml || export ERROR=true
   - sleep 5
-  - oc delete pods $(oc get pods -n default | grep mediawiki | awk $'{ print $1 }') -n default || export ERROR=true
+  - oc delete pods $(oc get pods -o name -l app=mediawiki123 -n default | head -1 | cut -f 2 -d '/') -n default
```
Comment (jmrodri) [inline .travis.yml]: good use of head and cut. I always forget about them.

File: Makefile [hunk 3]
```diff
@@ -60,4 +60,10 @@ deploy:
...
 	go test ./pkg/...
 
-.PHONY: vendor build run prepare-build-image build-image release-image release push clean deploy test
+cleanup-ci:
+	./scripts/broker-ci/cleanup-ci.sh
+
+ci:
+	./scripts/broker-ci/local-ci.sh
+
+.PHONY: vendor build run prepare-build-image build-image release-image release push clean deploy test ci cleanup-ci
```
Comment (jmrodri) [inline Makefile]: like the new targets. Do we want to add `cleanup-ci` to `really-clean`? Just curious maybe not needed.

File: README.md [hunk 4]
```diff
@@ -139,6 +139,19 @@ configurations.
...
+#### CI Target
+* `make ci`: Run the CI workflow that gets executed by travis, locally.
+   * Workflow:
+     - Provision Mediawiki
+     - Provision Postgresql
+     - Bind Postgresql and Mediawiki
+     - Curl the Mediawiki endpoint to check for success
+   * Requires:
+     - Cluster
+     - Service Catalog
```
Comment (jmrodri) [inline README.md]: Good addition.

File: scripts/broker-ci/cleanup-ci.sh [hunk 5]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+oc delete bindings.v1alpha1.servicecatalog.k8s.io mediawiki-postgresql-binding -n default
+./scripts/broker-ci/wait-for-resource.sh delete bindings.v1alpha1.servicecatalog.k8s.io mediawiki-postgresql-binding
+oc delete instance postgresql mediawiki -n default
+oc delete dc postgresql mediawiki123 -n default
+./scripts/broker-ci/wait-for-resource.sh delete pod postgresql
+./scripts/broker-ci/wait-for-resource.sh delete pod mediawiki
```
Comment (jmrodri) [inline scripts/broker-ci/cleanup-ci.sh]: +1 looks good to me

File: scripts/broker-ci/local-ci.sh [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    ./scripts/broker-ci/wait-for-resource.sh create pod mediawiki
+}
+
+function verify-ci-run {
+    ROUTE=$(oc get route -n default | grep mediawiki | cut -f 4 -d ' ')/index.php/Main_Page
+    BIND_CHECK=$(curl ${ROUTE}| grep "div class" | cut -f 2 -d "'")
+    echo "Running: curl ${ROUTE}| grep \"div class\" | cut -f 2 -d \"'\""
+    if [ "${BIND_CHECK}" = "error" ]; then
+	echo "MAKE CI FAILED"
+    elif [ "${BIND_CHECK}" = "" ]; then
```
Comment (jmrodri) [inline scripts/broker-ci/local-ci.sh]: Is it really `=` and not `==`?

File: scripts/broker-ci/local-ci.sh [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+	echo "Failed to gather data from ${ROUTE}. MAKE CI FAILED"
+    else
+	echo "SUCCESS"
+	echo "You can double check by opening http://${ROUTE} in your browser"
+    fi
+}
+
+provision
```
Comment (jmrodri) [inline scripts/broker-ci/local-ci.sh]: I'd prefer a comment above provision:

```
#
# MAIN
#
```

something like that just to make it easier to remember that this is where we start.

File: scripts/broker-ci/wait-for-resource.sh [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+    for r in $(seq 100); do
+	oc get ${RESOURCE} -n default | grep ${RESOURCE_NAME}
+	if [ $? -eq 0 ]; then
+	    echo "${RESOURCE_NAME} ${RESOURCE} has been created"
+	    break
+	fi
+	echo "Waiting for ${RESOURCE_NAME} ${RESOURCE} to be created"
+	sleep 1
+    done
+elif [ "${ACTION}" = "delete" ]; then
```
Comment (jmrodri) [inline scripts/broker-ci/wait-for-resource.sh]: +1 to rename, same question about `=` vs `==`.

File: .travis.yml (hunk 1 above)
Comment (rthallisey) [inline .travis.yml]: Yes. After make ci merges, travis is going to use make ci for testing. I'm going to follow up with that and some changes to make sure the scripts log really well and return good error messages.

File: Makefile (hunk 3 above)
Comment (rthallisey) [inline Makefile]: Sure.

File: scripts/broker-ci/local-ci.sh (hunk 6 above)
Comment (rthallisey) [inline scripts/broker-ci/local-ci.sh]: Either works.
```
string1 == string2
string1 = string2
       True if the strings are equal.
```

Comment (jmrodri) [inline scripts/broker-ci/local-ci.sh]: Thanks didn't know that.

**Pull request merged**